### PR TITLE
docker/install: Support rootless

### PR DESCRIPTION
**Pull request by vvoland**
Add support for running a rootless daemon. Currently only Linux host is supported.

### CODE REVIEW

**Comment by crazy-max** [comment]
https://github.com/docker/actions-toolkit/actions/runs/11687682100/job/32546382492?pr=485#step:10:198

```
  + echo rootlesskit needs to be installed
  rootlesskit needs to be installed
  + exit 1
```

Missing `rootlesskit`?

**Comment by crazy-max** [comment]
> It lost the PATH when `sudo`ing as a non-root user.

Might need `sudo -E`

**Comment by vvoland** [comment]
Ready for review @crazy-max

File: src/docker/install.ts [hunk 1]
```diff
@@ -339,21 +348,34 @@ export class Install {
...
     }) as {
       [key: string]: string;
     };
 
     await core.group('Start Docker daemon', async () => {
       const bashPath: string = await io.which('bash', true);
-      const cmd = `${this.toolDir}/dockerd --host="${dockerHost}" --config-file="${daemonConfigPath}" --exec-root="${this.runDir}/execroot" --data-root="${this.runDir}/data" --pidfile="${this.runDir}/docker.pid" --userland-proxy=false`;
+      let dockerPath = `${this.toolDir}/dockerd`;
+      if (this.rootless) {
+        dockerPath = `${this.toolDir}/dockerd-rootless.sh`;
```
Comment (crazy-max) [inline src/docker/install.ts:361]: I guess this could not work with `archive`?

Comment (vvoland) [reply src/docker/install.ts:361]: Hmmm right. We need rootless extras for this. Will try it.

Comment (vvoland) [reply src/docker/install.ts:365]: Done

**Review by crazy-max — APPROVED** [review]
LGTM thanks!

**Pull request merged**